app/factories/AuditEventFactory.scala

```diff
@@ -215,6 +215,29 @@ case class AuditEventFactory() {
     createSubmitGoodsRecordEvent(auditDetails)
   }
 
+  def createSubmitGoodsRecordEventForUpdateSupplementaryUnit(
+    affinityGroup: AffinityGroup,
+    journey: Journey,
+    supplementaryRequest: SupplementaryRequest,
+    recordId: String
+  )(implicit hc: HeaderCarrier): DataEvent = {
+    val auditDetails = Map(
+      "journey"              -> journey.toString,
+      "updateSection"        -> SupplementaryUnitUpdate.toString,
+      "recordId"             -> recordId,
+      "eori"                 -> supplementaryRequest.eori,
+      "affinityGroup"        -> affinityGroup.toString,
+      "addSupplementaryUnit" -> supplementaryRequest.hasSupplementaryUnit.getOrElse(false).toString,
+      "supplementaryUnit"    -> (if (supplementaryRequest.hasSupplementaryUnit.getOrElse(false)) {
```

Comment:
Should not set the field if supplementaryUnit is not set, rather than empty string it. 

"supplementaryUnit" -> supplementaryRequest.supplementaryUnit

should be fine because it shouldn't be set if hasSupplementaryUnit is not set anyway?

Reply:
we concatenate it with measurement unit so we need an if else